homework/data_warehousing_hw_2_KL.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Oct 11 21:16:17 2019
+
+@author: kristin.lomicka
+"""
+
+import dataset
+import pandas as pd
+import numpy as np
+
+#connect to postgres
+ds = dataset.connect("postgresql://postgres@localhost/hw1_data_warehouse")
+#export sql tables
+customers_a = pd.DataFrame(ds['customers_a'].all())
+employees_new = pd.DataFrame(ds['employees_new'].all())
+offices_a = pd.DataFrame(ds['offices_a'].all())
+order_metadata_a = pd.DataFrame(ds['order_metadata_a'].all())
+orders_new_a = pd.DataFrame(ds['orders_new_a'].all())
+products_a = pd.DataFrame(ds['products_a'].all())
+
+#create date dimension table
+date_d = order_metadata_a[['order_date']].drop_duplicates().reset_index(drop=True)
+date_d['order_date_time'] = pd.to_datetime(date_d['order_date'])
+date_d['order_date'] = pd.to_datetime(date_d['order_date']).dt.date
+date_d['day_of_week'] = date_d['order_date_time'].dt.weekday_name
+date_d['month'] = date_d['order_date_time'].dt.month
+date_d['year'] = date_d['order_date_time'].dt.year
+date_d['quarter']= date_d['order_date_time'].dt.quarter
+#insert unique ID
+date_d.insert(0, 'date_id', range(1, 1 + len(date_d)))
```

Comment:
This is a perfect thing to give to the database, which often takes care of unique id's! Postgres has a SERIAL type which is auto-incrementing.